compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -899,4 +901,40 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
 
         suggest_adding_lifetime_params(self.infcx.tcx, sub, ty_sup, ty_sub, diag);
     }
+
+    fn suggest_move_on_borrowing_closure(&self, diag: &mut Diagnostic) {
+        let map = self.infcx.tcx.hir();
+        let hir_id = self.mir_hir_id();
+        let body_id = map.body_owned_by(hir_id);
+        let expr = &map.body(body_id).value;
```

Comment:
This only works for functions that are of the form:

```
fn test() {
  || ...
}
```

or

```
fn test() {
  expr.method(|| ...)
```

This is not the right strategy, I think. It doesn't actually check that the closure here is part of the actual error, right?

Reply:
Thanks for reviewing. This suggest function will be only called when the closure has been identified having the error of borrowing problem, and we have to find where is the beginning of the problematic closure. So we are sure it is problematic closure at this point already. Yes, the current code will only find the span of the closure with above forms (in method, or in block). We can add more forms if any. But worst case is that it will not print the suggestion if the form is not supported yet. Does this makes sense?

Reply:
@compiler-errors Any comments/suggestions?